sickbeard/tvcache.py

```diff
@@ -32,98 +32,97 @@
+            if not self.hasColumn(provider_id, 'pubdate'):
+                self.addColumn(provider_id, 'pubdate', 'NUMERIC', '')
 
             # add hash column to table if missing
-            if not self.hasColumn(providerName, 'hash'):
-                self.addColumn(providerName, 'hash', "NUMERIC", "")
+            if not self.hasColumn(provider_id, 'hash'):
+                self.addColumn(provider_id, 'hash', 'NUMERIC', '')
 
         except Exception as e:
-            if str(e) != "table [" + providerName + "] already exists":
+            if str(e) != 'table [{0}] already exists'.format(provider_id):
                 raise
 
         # Create the table if it's not already there
         try:
             if not self.hasTable('lastUpdate'):
-                self.action("CREATE TABLE lastUpdate (provider TEXT, time NUMERIC)")
+                self.action('CREATE TABLE lastUpdate (provider TEXT, time NUMERIC)')
         except Exception as e:
-            logger.log(u"Error while searching " + self.provider.name + ", skipping: " + repr(e), logger.DEBUG)
+            logger.log(u'Error while searching {0}, skipping: {1!r}'.format(self.provider.name, e), logger.DEBUG)
             logger.log(traceback.format_exc(), logger.DEBUG)
-            if str(e) != "table lastUpdate already exists":
+            if str(e) != 'table lastUpdate already exists':
                 raise
 
 
 class TVCache(object):
     def __init__(self, provider, **kwargs):
         self.provider = provider
-        self.providerID = self.provider.get_id()
-        self.providerDB = None
+        self.provider_id = self.provider.get_id()
+        self.provider_db = None
         self.minTime = kwargs.pop(u'min_time', 10)
         self.search_params = kwargs.pop(u'search_params', dict(RSS=['']))
 
-    def _getDB(self):
+    def _get_db(self):
         # init provider database if not done already
-        if not self.providerDB:
-            self.providerDB = CacheDBConnection(self.providerID)
+        if not self.provider_db:
+            self.provider_db = CacheDBConnection(self.provider_id)
 
-        return self.providerDB
+        return self.provider_db
 
     def _clearCache(self):
         # Clear only items older than 7 days
         self._clearCacheItem()
-        #if self.shouldClearCache():
-        #    cache_db_con = self._getDB()
-        #    cache_db_con.action("DELETE FROM [" + self.providerID + "] WHERE 1")
 
     def _clearCacheItem(self):
-        cache_db_con = self._getDB()
+        cache_db_con = self._get_db()
         today = int(time.mktime(datetime.datetime.today().timetuple()))
         # Keep item in cache for 7 days
-        cache_db_con.action("DELETE FROM [" + self.providerID + "] WHERE time > ? ", [today + 7*86400]) # 86400 POSIX day (exact value)
+        cache_db_con.action('DELETE FROM [{0}] WHERE time < ? '.format(self.provider_id), [today - 7 * 86400])  # 86400 POSIX day (exact value)
```

Comment:
Good catch, however imo i don't care for enforcing a cache age of 7 days, particularly since we use the cache for manual search.  If anything this should be a configurable setting to allow the user how many days to trim, if the user wants to trim.